### PR TITLE
Add explanation about `Study.best_trials` in multi-objective optimization tutorial

### LINKED ISSUE #3421: Improve multi-objective optimization tutorial regarding `Study.best_trials`
## Background

https://github.com/optuna/optuna/issues/3397 added a note about `study.best_trials` to the [reuse the best trial tutorial](https://optuna.readthedocs.io/en/latest/tutorial/20_recipes/010_reuse_best_trial.html). During the discussion, we found that the [multi-objective optimization tutorial](https://optuna.readthedocs.io/en/latest/tutorial/20_recipes/002_multi_objective.html) does not fetch the trials on the Pareto front (i.e., `Study.best_trials`) while it visualize the Pareto front.

## Description

I think readers can understand how to use the trials on Pareto front more easily if we mention `Study.best_trials` in the tutorial page. We may have some possible scenarios. For example,

1. Show the Pareto front as can be seen in the current tutorial
2. Explain how to get the trials on the Pareto front by using `Study.best_trials`
3. Demonstrate how to get the desired trial from the Pareto front i.e., the trial whose accuracy is highest, and show the parameter values (or objective values)

image

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation

Resolve #3421.

## Description of the changes

Add explanation and examples about Study.best_trials in `tutorial/002_multi_objective.py`.

### CODE REVIEW

File: tutorial/20_recipes/002_multi_objective.py [hunk 1]
```diff
@@ -119,10 +119,23 @@ def objective(trial):
 
 
 ###################################################################################################
-# Check trials on pareto front visually
+# Check trials on pareto front visually.
```
Comment (toshihikoyanase) [inline tutorial/20_recipes/002_multi_objective.py:122]: A nit, but `Pareto` came from [`Vilfredo Pareto`](https://en.wikipedia.org/wiki/Vilfredo_Pareto) so please capitalize `p`.

```suggestion
# Check trials on Pareto front visually.
```

File: tutorial/20_recipes/002_multi_objective.py [hunk 2]
```diff
@@ -119,10 +119,23 @@ def objective(trial):
...
 optuna.visualization.plot_pareto_front(study, target_names=["FLOPS", "accuracy"])
 
 
+###################################################################################################
+# Fetch the list of trials on the Pareto front with :attr:`~optuna.study.Study.best_trials`.
+#
+# For example, the following code picks the trial with the highest accuracy from the Pareto front.
+
```
Comment (toshihikoyanase) [inline tutorial/20_recipes/002_multi_objective.py:130]: This section is rendered as follows:
image

IMHO, the description and code example may be a bit inconsistent, so could you add more words for it?
The following is just an example.

```suggestion
# You can fetch the list of trials on the Pareto front with :attr:`~optuna.study.Study.best_trials`.
#
# For example, the following code checks the number of the trials on the Pareto front and picks the trial with the highest accuracy.

```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR! The scenario of the code example seems great to me. I have some comments on the descriptions.

File: tutorial/20_recipes/002_multi_objective.py (hunk 1 above)
Comment (contramundum53) [reply tutorial/20_recipes/002_multi_objective.py]: Fixed.

File: tutorial/20_recipes/002_multi_objective.py (hunk 2 above)
Comment (contramundum53) [reply tutorial/20_recipes/002_multi_objective.py:130]: Thanks, I'll fix this.

Comment (contramundum53) [reply tutorial/20_recipes/002_multi_objective.py]: I fixed it in [8297474](https://github.com/optuna/optuna/pull/3443/commits/8297474e7bdce7f7d376cb0031d0073124668e03).

Comment (toshihikoyanase) [reply tutorial/20_recipes/002_multi_objective.py]: Thanks!

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**